pallets/gear/src/internal.rs

```diff
@@ -780,6 +818,26 @@ where
         });
     }
 
+    pub(crate) fn send_message_from_reservation(
+        message_id: MessageId,
+        source: ProgramId,
+        reservation_id: ReservationId,
+    ) {
+        GasHandlerOf::<T>::split(reservation_id, message_id)
+            .unwrap_or_else(|e| unreachable!("GasTree corrupted! {:?}", e));
+
+        Self::consume_gas_reservation(source, reservation_id);
+    }
+
+    pub(crate) fn consume_gas_reservation(program_id: ProgramId, reservation_id: ReservationId) {
```

Comment:
this seems more like delete_gas_reservation due to the fact the consumes goes inside in "remove_gas_reservation" fn